source/patterns/@aws-solutions-constructs/core/lib/sqs-helper.ts

```diff
@@ -125,13 +125,18 @@ export interface BuildDeadLetterQueueProps {
    * @default - Default props are used
    */
   readonly maxReceiveCount?: number
+  /**
+   * Whether this a first-in-first-out (FIFO) queue. (Must match source type)
+   *
+   */
+  readonly fifo?: boolean
 }
 
 export function buildDeadLetterQueue(scope: cdk.Construct, props: BuildDeadLetterQueueProps): sqs.DeadLetterQueue | undefined {
   if (!props.existingQueueObj && (props.deployDeadLetterQueue || props.deployDeadLetterQueue === undefined)) {
     // Create the Dead Letter Queue
     const [dlq] = buildQueue(scope, 'deadLetterQueue', {
-      queueProps: props.deadLetterQueueProps
+      queueProps: { ...props.deadLetterQueueProps, fifo: props.fifo ? true : undefined }
```

Comment:
Do you need this line? Line 132 already has already made fifo an attribute of deatLetterQueueProps.

Reply:
Yes, deadLetterQueueProps has the fifo property but this ensures it is passed properly depending on the logic:

Approach 1:
```
// Pass prop like this
this.deadLetterQueue = defaults.buildDeadLetterQueue(this, {
  existingQueueObj: props.existingQueueObj,
  deployDeadLetterQueue: props.deployDeadLetterQueue,
  deadLetterQueueProps: props.deadLetterQueueProps,
  maxReceiveCount: props.maxReceiveCount,
  fifo: props.queueProps && props.queueProps.fifo
});

// Build DLQ like this 
const [dlq] = buildQueue(scope, 'deadLetterQueue', {
      queueProps: { ...props.deadLetterQueueProps, fifo: props.fifo }
    });
```

OR 

Approach 2:
```
// Pass prop like this
this.deadLetterQueue = defaults.buildDeadLetterQueue(this, {
  existingQueueObj: props.existingQueueObj,
  deployDeadLetterQueue: { ...props.deployDeadLetterQueue, fifo: props.queueProps && props.queueProps.fifo },
  deadLetterQueueProps: props.deadLetterQueueProps,
  maxReceiveCount: props.maxReceiveCount
});

// Build DLQ like this 
const [dlq] = buildQueue(scope, 'deadLetterQueue', {
  queueProps: { ...props.deadLetterQueueProps }
});
```
I think the first approach is cleaner, what did you think?

Reply:
Manually tested before but now updated with integ and unit tests